src/android/toga_android/widgets/_pickers.py

```diff
@@ -0,0 +1,113 @@
+from ..libs.android.view import OnClickListener, View__MeasureSpec
+from ..libs.android.widget import EditText
+from .base import Widget
+
+
+class TogaPickerClickListener(OnClickListener):
+    def __init__(self, picker_impl):
+        super().__init__()
+        self.picker_impl = picker_impl
+
+    def onClick(self, _):
+        self.picker_impl._create_dialog()
+
+
+class TogaPickerSetListener:
+    def __init__(self, picker_impl):
+        super().__init__()
+        self.picker_impl = picker_impl
+
+    def listen(self, _, *args):
+        new_value = self.picker_impl._to_obj_converter_class(*self.picker_impl._value_pack_fn(*args))
+
+        self.picker_impl._showing = False
+        self.picker_impl._dialog = None
+        self.picker_impl.interface.value = new_value
+        if self.picker_impl.interface.on_change:
+            self.picker_impl.interface.on_change(self.picker_impl)
+
+    onDateSet = listen
+    onTimeSet = listen
+
+
+class PickerBase(Widget):
+    _icon = None
+    _hint = None
+    _to_obj_converter_class = None
+    _to_str_converter_kwargs = None
+    _dialog_class = None
+    _update_dialog_name = None
+    _value_unpack_fn = None
+    _value_pack_fn = None
+    _extra_dialog_setters = None
+    _extra_dialog_args = None
+    _dialog_listener_class = None
```

Comment:
Many of these method stub definitions are effectively reproducing what subclassing is for. 

For example, rather than defining `_to_obj_converter_class` and `_value_pack_fn`, we could define a method on the PickerBase  `to_obj()` that is given the args, and does the unpacking.

We can then declare a stub implementation in the base class that raises NotImplemented(), which is a more pythonic approach to ensuring subclasses implement all the required methods.

Reply:
Made everything as methods (mostly "private" with a single underscore), so this is removed. On top of that, the methods are defined as abstract methods, so they will throw an error if not overridden. I personally am not a  big fan of this way of implementing, since most of these just return a single, static value, but here you go :)